crates/oxc_linter/src/service/runtime.rs

```diff
@@ -206,6 +206,8 @@ impl Runtime {
         // The semantic model is not built at this stage.
         let semantic_builder = SemanticBuilder::new()
             .with_cfg(true)
+            // TODO: this is only used by exhaustive_deps, consider conditionally enabling?
+            .with_scope_tree_child_ids(true)
```

Comment:
I gave it a shot, I think this works:

```rust
let should_build_child_ids = self.linter.rules.iter().any(|rule| matches!(rule.rule, RuleEnum::ExhaustiveDeps(_)));
```

```suggestion
            .with_scope_tree_child_ids(should_build_child_ids)
```

Reply:
hmm let me get the benchmark working and look at the perf hit (if any)

Reply:
<img width="1130" alt="Screenshot 2024-11-09 at 15 58 35" src="https://github.com/user-attachments/assets/794d48c5-ab73-4527-ab11-6a34797989e0">

shall we leave for now? look like there's no perf change


(i'm just about to put a comit to disable running this rule on non-`tsx/jsx` files to fix the checker.ts perf regression)

Reply:
It might just be random, that doesn't seem like a big perf change (if any) to me so I think it's fine